sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCheckResult.scala

```diff
@@ -35,14 +35,6 @@ object TypeCheckResult {
     def isSuccess: Boolean = true
   }
 
-  /**
-   * Represents the failing result of `Expression.checkInputDataTypes`,
-   * with an error message to show the reason of failure.
-   */
-  case class TypeCheckFailure(message: String) extends TypeCheckResult {
```

Comment:
@MaxGekk Should we mark this as `@deprecated` in Spark 3.4.0 and explicitly declare that it will be deleted in Spark 4.0( or other version)?

Reply:
This is an internal API. We deprecate a public API usually. @cloud-fan WDYT?

Reply:
hmm... although other projects rely on it, I think it's OK to delete it now